lib/master_manipulator/service.rb

```diff
@@ -2,71 +2,226 @@
+    # *+silent+ - Suppress beaker's output; set to false to see it
+    #
+    # ==== Examples
+    # encode_cert(master, '/etc/ssl/private/host.cert')
+    # encode_cert(master, '/etc/ssl/private/host.cert', false)
+    def encode_cert(cert_host, cert_file, silent = true)
+      rawcert = on(cert_host, "cat #{cert_file}", {:silent => silent}).stdout.strip
+      OpenSSL::X509::Certificate.new(rawcert)
+    end
 
-      message = "Attempted to restart #{opts[:wait_cycles]} times, waited #{total_time} seconds total."
+    # Convert the contents of the private key file in key_file on the host
+    # specified by key_host into an RSA private key and return it
+    # ==== Attributes
+    # *+key_host+ - The host whose key you want
+    # *+key_file+ - The specific key file you want
+    # *+silent+ - Suppress beaker's output; set to false to see it
+    #
+    # ==== Examples
+    # encode_key(master, '/etc/ssl/private/host.key')
+    # encode_key(agent, '/etc/ssl/private/host.key', :silent => false)
+    def encode_key(key_host, key_file, silent = true)
+      rawkey = on(key_host, "cat #{key_file}", {:silent => silent}).stdout.strip
+      OpenSSL::PKey::RSA.new(rawkey)
+    end
 
-      message << "\nHere is the status reported by the puppetserver'"
+    # Issue an HTTP request and return the Net::HTTPResponse object. Lifted from
+    # pe_acceptance_tests/2015.3.x/lib/http_calls.rb and slightly modified.
+    # ==== Attributes
+    # **url+ - (String) URL to poke
+    # **method+ - (Symbol) :post, :get
+    # **cert+ - (OpenSSL::X509::Certificate, String) The certificate to
+    #       use for authentication.
+    # **key+ - (OpenSSL::PKey::RSA, String) The private key to use for
+    #      authentication
+    # **body+ - (String) Request body (default empty)
+    #
+    # ==== Examples
+    require 'net/http'
+    require 'uri'
+    def https_request(url, request_method, cert, key, body = nil)
+      # Make insecure https request
+      uri = URI.parse(url)
+      http = Net::HTTP.new(uri.host, uri.port)
+
+      if cert.is_a?(OpenSSL::X509::Certificate)
+        http.cert = cert
+      else
+        raise TypeError, 
+          "cert must be an OpenSSL::X509::Certificate object, not #{cert.class}"
+      end
 
-      @body.each do |k,v|
-        message << "\n'#{k}' state: #{v['state']} "
+      if key.is_a?(OpenSSL::PKey::RSA)
+        http.key = key
+      else
+        raise TypeError,
+        "key must be an OpenSSL::PKey:RSA object, not #{key.class}"
       end
 
-      raise message
+      http.use_ssl = true
+      http.verify_mode = OpenSSL::SSL::VERIFY_NONE
 
+      if request_method == :post
+        request = Net::HTTP::Post.new(uri.request_uri)
+        # Needs the content type even though no data is being sent
+        request.content_type = 'application/json'
+        request.body = body
+      else
+        request = Net::HTTP::Get.new(uri.request_uri)
+      end
+      response = http.request(request)
     end
```

Comment:
if it goes into scooter it should be modified to pick up the changes from pe_acceptance_tests, specifically around support for `Put` and `Delete` requests.